### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Limits.KanExtension

### PR DESCRIPTION
**Pull request by adamtopaz**
---
- [x] depends on: #2545

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2545
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2545~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Limits/KanExtension.lean [hunk 1]
```diff
@@ -65,81 +66,112 @@ def cone {F : S ⥤ D} {G : L ⥤ D} (x : L) (f : ι ⋙ G ⟶ F) : Cone (diagra
...
 
 end Ran
 
 /-- The right Kan extension of a functor. -/
-@[simps]
+@[simps!]
 def ran [∀ X, HasLimitsOfShape (StructuredArrow X ι) D] : (S ⥤ D) ⥤ L ⥤ D :=
-  Adjunction.rightAdjointOfEquiv (fun F G => (Ran.equiv ι G F).symm) (by tidy)
-#align category_theory.Ran CategoryTheory.ran
+  Adjunction.rightAdjointOfEquiv (fun F G => (Ran.equiv ι G F).symm) (by {
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/KanExtension.lean:166]:
```suggestion
  Adjunction.rightAdjointOfEquiv (fun F G => (Ran.equiv ι G F).symm) (by {
    -- Porting note: was `tidy`
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**